library/core/src/intrinsics.rs

```diff
@@ -2211,7 +2211,9 @@ macro_rules! assert_unsafe_precondition {
             fn runtime$(<$($tt)*>)?($($i:$ty),*) {
                 if !$e {
                     // don't unwind to reduce impact on code size
-                    ::core::panicking::panic_str_nounwind("unsafe precondition violated");
+                    ::core::panicking::panic_str_nounwind(
+                        concat!("unsafe precondition violated: ", stringify!($e))
```

Comment:
This will lead to quite ugly errors, won't it?

IMO we should have a custom string argument in the macro so use sites have say something user-understandable here.

Reply:
I just added the name of the functions the assertions are for. I'm bothered by the maintenance burden, I'd like to have something that's handled automatically which is the reason why I was trying to stringify the passed expression.

I'm thinking about how to word the rest of the panic messages.

Reply:
Okay, I took a shot at improving a few of the error messages with some English. Thoughts? I feel like I might be spending too much time agonizing over the wording.

Reply:
But I guess as a start I am also fine with just stringifying the expression :shrug: